src/test/java/jenkins/branch/CustomOrganizationFolderDescriptorTest.java

```diff
@@ -76,35 +76,41 @@ public void removeIntegrationTestMocks() throws Exception {
     
     @Test
     public void noNavigatorNoFactoryInstalled() throws Exception {
-        assertEquals(1, ExtensionList.lookup(SCMNavigatorDescriptor.class).size());
-        assertEquals(SingleSCMNavigator.DescriptorImpl.class, ExtensionList.lookup(SCMNavigatorDescriptor.class).get(0).getClass());
-        assertEquals(0, ExtensionList.lookup(MultiBranchProjectFactoryDescriptor.class).size());
+        assertThat(ExtensionList.lookup(SCMNavigatorDescriptor.class),
+                hasItem(ExtensionList.lookupSingleton(SingleSCMNavigator.DescriptorImpl.class)));
+        // Ensure none of our mock factories sneaked into this test (though other MultiBranchProjectFactory impls may exist on the classpath)
+        assertThat(ExtensionList.lookup(OrganizationFolderTest.MockFactoryDescriptor.class), is(empty()));
         assertThat(newItemTypes(), containsInAnyOrder());
```

Comment:
This is going to fail in PCT if something like `github-branch-source` is included, I think. Would need to be amended to assert that none of the mock types are listed. Could be done later perhaps.

Reply:
does `github-branch-source` have classes extending OrganizationFolderTest.MockFactoryDescriptor? if it did, it would have to be depending on inner test classes from this plugin - something not recommended and probably unlikely.

Reply:
No, I am referring to the `newItemTypes` being empty.

Reply:
(or did you mean there might be classes extending the navigator?)

Reply:
If I am remembering things right, this last line (which should probably have read `is(empty())` for clarity) is asserting that if you actually went to the **New Item** page, you would not be offered any organization folder types.

I wonder if this PR is getting too complicated. Might be easier to just `assumeThat` the various extension lists (such as of `SCMNavigatorDescriptor`) do not contain any implementations outside this plugin’s normal test classpath—in other words, just skip these tests in PCT mode because they would not make sense.

Reply:
If that's an acceptable outcome then yeah it would be the easier option.

These tests clearly have assumptions about what their world looks like - assumptions which would be reasonable in most Java projects where the tests are run the straightforward way. Generalising the assertions to account for unknowable interlopers on the classpath will definitely be more complicated.

Reply:
Yes let us just use `Assume` to skip some of these tests in PCT, since their topic is really what happens when other stuff is _not_ installed. Sorry for not suggesting this earlier.

Reply:
have reverted CustomOrganizationFolderDescriptorTest to master and then mechanically ported the asserts to assumeThat.

Shall I do the same for the other tests?

Reply:
Yes, that is looking much more straightforward.

Reply:
I think the other tests already passed in PCT though, right?

Reply:
yep they did, so we could leave them as is.